src/kbmod_wf/task_impls/reproject_wu.py

```diff
@@ -12,3 +23,289 @@ def reproject_wu(input_wu=None, reprojected_wu=None, logger=None):
+        )
+
+        return self.reprojected_wu_filepath
+
+    def _get_params_from_uri_file(self):
+        """
+        Get parameters we place into URI file as comments at the top.
+        Example start of URI file (6/6/2024 COC):
+        #desired_dates=['2019-04-02', '2019-05-07']
+        #dist_au=42.0
+        #patch_size=[20, 20]
+        #patch_id=5845
+        #patch_center_coords=(216.49999999999997, -13.500000000000005)
+        #patch_box=[[216.33333333333331, -13.666666666666671], [216.33333333333331, -13.333333333333337], [216.66666666666666, -13.333333333333337], [216.66666666666666, -13.666666666666671], [216.33333333333331, -13.666666666666671]]
+        /gscratch/dirac/DEEP/repo/DEEP/20190507/A0b/science#step6/20240425T145342Z/differenceExp/20190508/VR/VR_DECam_c0007_6300.0_2600.0/855719/differenceExp_DECam_VR_VR_DECam_c0007_6300_0_2600_0_855719_S12_DEEP_20190507_A0b_scienceHASHstep6_20240425T145342Z.fits
+        6/6/2024 COC
+        """
+        results = {}
+        with open(self.uri_filepath, "r") as f:
+            for line in f:
+                line = line.strip()
+                if line == "":
+                    continue  # deal with rogue blank lines or invisible double line endings
+                if not line.startswith("#"):
+                    break  # comments section is done
+                line = line.lstrip("#").split("=")
+                lhs = line[0].strip()
+                rhs = line[1].strip()
+                try:
+                    rhs = eval(rhs)
+                except ValueError:
+                    self.logger.debug(f"Unable to eval() {lhs} field with value {rhs}.")
+                    continue
+                results[lhs] = rhs
+        return results
+
+    def _patch_arcmin_to_pixels(self, patch_size_arcmin, pixel_scale_arcsec_per_pix):
+        """
+        Take an array of two dimensions, in arcminutes, and convert this to pixels using the supplied pixel scale (in arcseconds per pixel).
+        6/6/2024 COC
+        """
+        x_pixels = int(np.ceil((patch_size_arcmin[0] * 60) / pixel_scale_arcsec_per_pix))
+        y_pixels = int(np.ceil((patch_size_arcmin[1] * 60) / pixel_scale_arcsec_per_pix))
+
+        # ! TODO, can't the above lines be changed to the following?
+        # ! And thereby eliminate the to input parameters to the function?
+        # patch_pixels = int(np.ceil(self.patch_size * 60 / self.pixel_scale))
+
+        patch_pixels = [x_pixels, y_pixels]
+
+        self.logger.debug(
+            f"Derived patch_pixels (w, h) = {patch_pixels} from patch_size_arcmin={patch_size_arcmin} and pixel_scale_arcsec_per_pix={pixel_scale_arcsec_per_pix}."
+        )
+
+        return x_pixels, y_pixels
+
+    def _create_wcs_from_corners(
+        self,
+        corners=None,
+        image_width=None,
+        image_height=None,
+        filename="output.fits",
+        pixel_scale=None,
+        verbose=True,
+    ):
```

Comment:
@wilsonbb Just FYI, I just didn't update this method, there's no particular reason. But I don't think that any of the input parameters actually need to be passed in, they should all be class members, and can be referenced inside this method by `self.<var_name>`. 

If you're excited about jumping in on this PR, I wouldn't mind the help cleaning this one up :)

Reply:
Yeah that makes sense to me. I think we might be able to replace most of this function with a call to https://docs.astropy.org/en/stable/api/astropy.wcs.utils.fit_wcs_from_points.html

which should make the above easier. But I am fine submitting this PR with the current method as is and revisiting on the next pass.

Reply:
I'll leave this as is, and we can update as needed.